### ISSUE TITLE
Not installable on macOS

### ISSUE THREAD

**Issue by philippmayrth** [tree-sitter/py-tree-sitter#66]
I cant install this on macOS with the following error message. I tried with several Python versions and with pip3 and with pipenv to see if there is a difference.
I also updated pip3, setuptools and wheel.

I did get it working on Ubuntu 20.04 with Python 3.8.

Error message on macOS.
```
➜  NO pipenv install tree-sitter --python 3.8
Virtualenv already exists!
Removing existing virtualenv...
Creating a virtualenv for this project...
Pipfile: /Users/zensored/Desktop/NO/Pipfile
Using /usr/bin/python3 (3.8.2) to create virtualenv...
⠧ Creating virtual environment...created virtual environment CPython3.8.2.final.0-64 in 396ms
  creator CPython3macOsFramework(dest=/Users/zensored/.local/share/virtualenvs/NO--wY_wDPi, clear=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/Users/zensored/Library/Application Support/virtualenv)
    added seed packages: pip==20.2.4, setuptools==50.3.2, wheel==0.35.1
  activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator

✔ Successfully created virtual environment! 
Virtualenv location: /Users/zensored/.local/share/virtualenvs/NO--wY_wDPi
Installing tree-sitter...
Error:  An error occurred while installing tree-sitter!
Error text: Collecting tree-sitter
  Using cached tree_sitter-0.19.0.tar.gz (112 kB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
    Preparing wheel metadata: started
    Preparing wheel metadata: finished with status 'done'
Building wheels for collected packages: tree-sitter
  Building wheel for tree-sitter (PEP 517): started
  Building wheel for tree-sitter (PEP 517): finished with status 'error'
Failed to build tree-sitter

  ERROR: Command errored out with exit status 1:
   command: /Users/zensored/.local/share/virtualenvs/NO--wY_wDPi/bin/python /Users/zensored/.local/share/virtualenvs/NO--wY_wDPi/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py build_wheel /var/folders/ww/9g40xcr51854gq69rrk2smt80000gn/T/tmpzxxy5bw0
       cwd: /private/var/folders/ww/9g40xcr51854gq69rrk2smt80000gn/T/pip-install-40n0w3yk/tree-sitter
  Complete output (26 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-10.14.6-x86_64-3.8
  creating build/lib.macosx-10.14.6-x86_64-3.8/tree_sitter
  copying tree_sitter/__init__.py -> build/lib.macosx-10.14.6-x86_64-3.8/tree_sitter
  warning: build_py: byte-compiling is disabled, skipping.
  
  running build_ext
  building 'tree_sitter.binding' extension
  creating build/temp.macosx-10.14.6-x86_64-3.8
  creating build/temp.macosx-10.14.6-x86_64-3.8/tree_sitter
  creating build/temp.macosx-10.14.6-x86_64-3.8/tree_sitter/core
  creating build/temp.macosx-10.14.6-x86_64-3.8/tree_sitter/core/lib
  creating build/temp.macosx-10.14.6-x86_64-3.8/tree_sitter/core/lib/src
  clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -iwithsysroot/System/Library/Frameworks/System.framework/PrivateHeaders -iwithsysroot/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/Headers -arch arm64 -arch x86_64 -Itree_sitter/core/lib/include -Itree_sitter/core/lib/src -I/Users/zensored/.local/share/virtualenvs/NO--wY_wDPi/include -I/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/include/python3.8 -c tree_sitter/core/lib/src/lib.c -o build/temp.macosx-10.14.6-x86_64-3.8/tree_sitter/core/lib/src/lib.o -std=c99 -Wno-unused-variable
  clang: warning: using sysroot for 'iPhoneSimulator' but targeting 'MacOSX' [-Wincompatible-sysroot]
  clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -iwithsysroot/System/Library/Frameworks/System.framework/PrivateHeaders -iwithsysroot/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/Headers -arch arm64 -arch x86_64 -Itree_sitter/core/lib/include -Itree_sitter/core/lib/src -I/Users/zensored/.local/share/virtualenvs/NO--wY_wDPi/include -I/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/include/python3.8 -c tree_sitter/binding.c -o build/temp.macosx-10.14.6-x86_64-3.8/tree_sitter/binding.o -std=c99 -Wno-unused-variable
  clang: warning: using sysroot for 'iPhoneSimulator' but targeting 'MacOSX' [-Wincompatible-sysroot]
  clang -bundle -undefined dynamic_lookup -Wl,-headerpad,0x1000 -arch arm64 -arch x86_64 build/temp.macosx-10.14.6-x86_64-3.8/tree_sitter/core/lib/src/lib.o build/temp.macosx-10.14.6-x86_64-3.8/tree_sitter/binding.o -o build/lib.macosx-10.14.6-x86_64-3.8/tree_sitter/binding.cpython-38-darwin.so
  clang: warning: using sysroot for 'iPhoneSimulator' but targeting 'MacOSX' [-Wincompatible-sysroot]
  ld: warning: -undefined dynamic_lookup is deprecated on iOS Simulator
  ld: -platform_version passed unknown platform name 'macos-simulator'
  clang: error: linker command failed with exit code 1 (use -v to see invocation)
  error: command 'clang' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for tree-sitter
ERROR: Could not build wheels for tree-sitter which use PEP 517 and cannot be installed directly

✘ Installation Failed 

```

macOS 11.1 (20C69)
XCode Version 12.4 (12D4e)
Python 3.8, 3.9, 3.10 (yes I tested them all)
pip 21.0.1 from /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pip (python 3.10)
pipenv, version 2020.11.15

**Comment by lunixbochs** [comment]
> clang: warning: using sysroot for 'iPhoneSimulator' but targeting 'MacOSX' [-Wincompatible-sysroot]

It's working fine for me, and working in CI. Seems like something wrong with your environment.

What does `xcrun --show-sdk-path` show for you? What's your output for the `env` command in your shell?

**Comment by philippmayrth** [comment]
It shows /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk but if it works for others feel free to close this ticket. I already got it working on a different machine so I don't need it working on my Mac.

**Comment by lunixbochs** [comment]
Are you able to build _any_ compiled python extensions in this environment?

**Comment by memray** [comment]
Failed to install on MacOS either.

**Comment by memray** [comment]
Installing latest Python env resolves the issue (my case 3.8.8->3.8.16).

**Comment by ekg15** [comment]
Also works w/ 3.10
